src/core/dfn-contract.js

```diff
@@ -0,0 +1,61 @@
+export const name = "core/dfn-contract";
+
+export function run() {
+  addContractDefaults();
+  addDefinitionPointers();
+}
+
+function addContractDefaults() {
+  // Find all dfns that don't have a type and default them to "dfn".
+  /** @type NodeListOf<HTMLElement> */
+  const dfnsWithNoType = document.querySelectorAll(
+    "dfn:is([data-dfn-type=''],:not([data-dfn-type]))"
+  );
+  for (const dfn of dfnsWithNoType) {
+    dfn.dataset.dfnType = "dfn";
+  }
+
+  // Per "the contract", export all definitions, except where:
```

Comment:
This doesn't seem right... we should only export definitions explicitly marked as export, no?